src/Demo/main.c

```diff
@@ -0,0 +1,182 @@
+/**@file
+ * This file is part of the Leon3 BSP for the Test Environment.
+ *
+ * @copyright 2022-2023 N7 Space Sp. z o.o.
+ *
+ * Test Environment was developed under a programme of,
+ * and funded by, the European Space Agency (the "ESA").
+ *
+ *
+ * Licensed under the ESA Public License (ESA-PL) Permissive,
+ * Version 2.3 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://essr.esa.int/license/list
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// \brief Example program for UART interface
+
+#include "Uart.h"
+#include "UartRegisters.h"
+#include "ByteFifo.h"
+#include "SystemConfig.h"
+#include "Startup_Leon3.h"
+#include <rtems.h>
+
+#define REG_BIT(reg, n) (((reg) >> n) & 1U)
+#define REG_BYTE(reg, n) (((reg) >> (n * 4)) & 0x0FU)
```

Comment:
GET_BYTE? (this does not have to be a register....)

Reply:
removed (changed method of code testing)